lib/_http_common.js

```diff
@@ -43,22 +23,11 @@ function parserOnHeadersComplete(versionMajor, versionMinor, headers, method,
                                  url, statusCode, statusMessage, upgrade,
                                  shouldKeepAlive) {
   var parser = this;
-
-  if (!headers) {
-    headers = parser._headers;
-    parser._headers = [];
-  }
-
-  if (!url) {
-    url = parser._url;
-    parser._url = '';
-  }
-
-  parser.incoming = new IncomingMessage(parser.socket);
-  parser.incoming.httpVersionMajor = versionMajor;
-  parser.incoming.httpVersionMinor = versionMinor;
-  parser.incoming.httpVersion = versionMajor + '.' + versionMinor;
-  parser.incoming.url = url;
+  var stream = parser.incoming = new IncomingMessage(parser.socket);
```

Comment:
I just reverted that portion and I couldn't really tell any difference since the `http/simple` benchmark numbers can bounce around some every time. If there is a difference, it's probably minor. \*shrug\*